tests/auto_generated/SimplicialComplex_test_autogen.py

```diff
@@ -0,0 +1,126 @@
+from SimplicialComplex import *
+import random
+
+def write_cpp_headers(file):
+    cpp_headers = """
+    #include <catch2/catch_test_macros.hpp>
+    #include <wmtk/SimplicialComplex.hpp>
+    #include <wmtk/TetMesh.hpp>
+    #include <wmtk/TriMesh.hpp>
+    #include "../tools/DEBUG_TriMesh.hpp"
+    #include <igl/read_triangle_mesh.h>
+
+    using namespace wmtk;
+
+    constexpr PrimitiveType PV = PrimitiveType::Vertex;
+    constexpr PrimitiveType PE = PrimitiveType::Edge;
+    constexpr PrimitiveType PF = PrimitiveType::Face;
+    constexpr PrimitiveType PT = PrimitiveType::Tetrahedron;
+    """
+    file.write(cpp_headers)
+
+def get_test_function(mesh, test_function_name):
+    # Choose test function
+    if test_function_name == 'open_star':
+        return mesh.open_star
+    elif test_function_name == 'closed_star':
+        return mesh.closed_star
+    elif test_function_name == 'link':
+        return mesh.link
+    elif test_function_name == "simplex_with_boundary":
+        return mesh.boundary
+    elif test_function_name == "top_coface_simplex":
+        return mesh.top_coface_simplex
+    else:
+        raise Exception("Function name not recognized")
+    
+def write_tests(file, path_to_data, mesh_name, n_samples, test_function_names):
+    mesh = load_obj(path_to_data + mesh_name)
```

Comment:
there needs to also be functionality for randomly deleting faces and and checkign that the result of those deletions is a manifold mesh